### PR TITLE
add 2d tests to ci

### PR DESCRIPTION
**Pull request by wanchaol**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #96700
* __->__ #96711

### CODE REVIEW

**Review by huydhn — APPROVED** [review]

File: .ci/pytorch/multigpu-test.sh [hunk 1]
```diff
@@ -39,6 +39,10 @@ time python test/run_test.py --verbose -i distributed/_shard/sharded_tensor/ops/
 time python test/run_test.py --verbose -i distributed/_shard/sharded_tensor/ops/test_matrix_ops
 time python test/run_test.py --verbose -i distributed/_shard/sharded_tensor/ops/test_softmax
 time python test/run_test.py --verbose -i distributed/_shard/sharded_optim/test_sharded_optim
+
+# DTensor/TP tests
+time python test/run_test.py --verbose -i distributed/tensor/parallel/test_2d_parallel
+
```
Comment (fduwjj) [inline .ci/pytorch/multigpu-test.sh:45]: Can we also add test_tp_example? Thanks!

**Review by fduwjj — APPROVED** [review]
Thanks for adding this into CI!!!

**Comment by wanchaol** [comment]
Looks like the ci failures are not related. landing

**Pull request closed without merging**